public/app/features/plugins/PluginListPage.tsx

```diff
@@ -20,40 +23,49 @@ export interface Props {
   setPluginsSearchQuery: typeof setPluginsSearchQuery;
 }
 
-export class PluginListPage extends PureComponent<Props> {
-  componentDidMount() {
-    this.fetchPlugins();
-  }
-
-  async fetchPlugins() {
-    await this.props.loadPlugins();
-  }
+export const PluginListPage: React.FC<Props> = ({
+  hasFetched,
+  navModel,
+  plugins,
+  setPluginsSearchQuery,
+  searchQuery,
+  loadPlugins,
+}) => {
+  useAsync(async () => {
+    return await loadPlugins();
+  }, [loadPlugins]);
 
-  render() {
-    const { hasFetched, navModel, plugins, setPluginsSearchQuery, searchQuery } = this.props;
+  const linkButton = {
+    href: 'https://grafana.com/plugins?utm_source=grafana_plugin_list',
+    title: 'Find more plugins on Grafana.com',
+  };
 
-    const linkButton = {
-      href: 'https://grafana.com/plugins?utm_source=grafana_plugin_list',
-      title: 'Find more plugins on Grafana.com',
-    };
+  return (
+    <Page navModel={navModel} aria-label={selectors.pages.PluginsList.page}>
+      <Page.Contents isLoading={!hasFetched}>
+        <>
+          <OrgActionBar
+            searchQuery={searchQuery}
+            setSearchQuery={query => setPluginsSearchQuery(query)}
+            linkButton={linkButton}
+            target="_blank"
+          />
 
-    return (
-      <Page navModel={navModel}>
-        <Page.Contents isLoading={!hasFetched}>
-          <>
-            <OrgActionBar
-              searchQuery={searchQuery}
-              setSearchQuery={query => setPluginsSearchQuery(query)}
-              linkButton={linkButton}
-              target="_blank"
-            />
-            {hasFetched && plugins && plugins && <PluginList plugins={plugins} />}
-          </>
-        </Page.Contents>
-      </Page>
-    );
-  }
-}
+          <PluginsErrorsInfo>
+            <>
+              <br />
+              <p>
+                Note that <strong>unsigned front-end datasource and panel plugins</strong> are still usable, but this is
+                a subject of change in the upcoming releases of Grafana
```

Comment:
```suggestion
                subject to change in the upcoming releases of Grafana
```

Reply:
Also this is a second time this text/component is used, so maybe it's better to extract this?

Reply:
it's a different text ;)

Reply:
Ah, you're right, noticed the subtle difference 🙈